Panic with Go 1.10 With Deoplete-Go and NVim

I'm having a lot of issues with gocode panicking. I'll have freshly opened Vim in a Go project and start typing. Neovim will get flooded with messages about how gocode's panicked.

Here are the logs from the debug:

```
panic: runtime error: slice bounds out of range

goroutine 28 [running]:
runtime/debug.Stack(0xb, 0xc42005e820, 0x1)
	/usr/local/Cellar/go/1.10.3/libexec/src/runtime/debug/stack.go:24 +0xa7
runtime/debug.PrintStack()
	/usr/local/Cellar/go/1.10.3/libexec/src/runtime/debug/stack.go:16 +0x22
main.(*Server).AutoComplete.func1(0xc42022e340)
	/Users/catherinecai/go/src/github.com/mdempsky/gocode/server.go:72 +0x94
panic(0x141cf60, 0x16fab90)
	/usr/local/Cellar/go/1.10.3/libexec/src/runtime/panic.go:502 +0x229
main.(*Server).AutoComplete(0x172a070, 0xc42018ac30, 0xc42022e340, 0x0, 0x0)
	/Users/catherinecai/go/src/github.com/mdempsky/gocode/server.go:84 +0x9ac
reflect.Value.call(0xc4201c4000, 0xc420168038, 0x13, 0x149d878, 0x4, 0xc42005ef18, 0x3, 0x3, 0xc42015e2c0, 0x13f65e0, ...)
	/usr/local/Cellar/go/1.10.3/libexec/src/reflect/value.go:447 +0x969
reflect.Value.Call(0xc4201c4000, 0xc420168038, 0x13, 0xc42004cf18, 0x3, 0x3, 0x14688a0, 0x147f201, 0xc42015f440)
	/usr/local/Cellar/go/1.10.3/libexec/src/reflect/value.go:308 +0xa4
net/rpc.(*service).call(0xc42016e0c0, 0xc420178050, 0xc42017ee48, 0xc42017ee60, 0xc42017c080, 0xc420164180, 0x13e24c0, 0xc42018ac30, 0x16, 0x13e2480, ...)
	/usr/local/Cellar/go/1.10.3/libexec/src/net/rpc/server.go:384 +0x14e
created by net/rpc.(*Server).ServeCodec
	/usr/local/Cellar/go/1.10.3/libexec/src/net/rpc/server.go:480 +0x43a
2018/07/21 11:36:21 Got autocompletion request for '/Users/catherinecai/go/src/gitlab.com/catherinetcai/rekognize/server/server/face.go'
2018/07/21 11:36:21 Cursor at: 190
2018/07/21 11:36:21 -------------------------------------------------------
```

What I'm running:

Go: go version go1.10.3 darwin/amd64
Neovim Version: 
```
NVIM v0.3.0
Build type: Release
LuaJIT 2.0.5
Compilation: /usr/local/Homebrew/Library/Homebrew/shims/mac/super/clang -Wconversion -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=1 -DNVIM_MSGPACK_HAS_FLOAT32 -DNVIM_UNIBI_HAS_VAR_FROM -DNDEBUG -DMIN_LOG_LEVEL=3 -Wall -Wextra -pedantic -Wno-unused-parameter -Wstrict-prototypes -std=gnu99 -Wimplicit-fallthrough -Wvla -fstack-protector-strong -fdiagnostics-color=auto -DINCLUDE_GENERATED_DECLARATIONS -I/tmp/neovim-20180612-71518-1f95nfa/neovim-0.3.0/build/config -I/tmp/neovim-20180612-71518-1f95nfa/neovim-0.3.0/src -I/usr/local/include -I/usr/local/opt/gettext/include -I/usr/include -I/tmp/neovim-20180612-71518-1f95nfa/neovim-0.3.0/build/src/nvim/auto -I/tmp/neovim-20180612-71518-1f95nfa/neovim-0.3.0/build/include
Compiled by brew@HighSierra.local
```

Not sure where to go next with this...

--- a/internal/lookdot/lookdot.go
+++ b/internal/lookdot/lookdot.go
@@ -74,7 +74,7 @@ func walk(typ0 types.Type, addable0, value bool, v Visitor) {
 			for _, t := range next {
 				nt := namedOf(t.typ)
 				if nt == nil {
-					panic("embedded struct field without name?")
+					continue
 				}
 				if !visited[nt] {
 					cur = append(cur, t)

